### ISSUE TITLE
Stylelint not reporting any errors on Preact project

### ISSUE THREAD

**Issue by lipe-dev** [stylelint/stylelint#5755]
### What steps are needed to reproduce the bug?

Running stylelint against any files doesn't generate any errors, nor does --fix do anything.

This is the same config I use in all my other projects, and they work fine there.

I can reproduce it without the order and prettier plugins as well, I can basically write any CSS and it will allow it (for either tsx or .css files)

### What Stylelint configuration is needed to reproduce the bug?

```js
const sortOrderSmacss = require("stylelint-config-property-sort-order-smacss/generate");

module.exports = {
  extends: [
    "stylelint-config-recommended",
    "stylelint-config-property-sort-order-smacss",
    "stylelint-config-prettier",
  ],
  plugins: ["stylelint-order"],
  customSyntax: "@stylelint/postcss-css-in-js",
  rules: {
    "order/properties-order": [
      sortOrderSmacss({ noEmptyLineBetween: true, emptyLineBefore: "always" }),
    ],
  },
  overrides: [
    {
      files: ["**/*.{css,md,msx,js,jsx,ts,tsx,json,yml,yaml}"],
    },
  ],
};
```

### How did you run Stylelint?

`yarn stylelint src/components/navbar/index.tsx` and also `yarn stylelint src/style/index.css`

### Which version of Stylelint are you using?

14.1.0

### What did you expect to happen?

Some errors should be reported (I basically broke the CSS code in every way I could imagine)

### What actually happened?

Nothing. It executes and finishes without any errors as if the files were OK.

### Does the bug relate to non-standard syntax?

I tried it both with (styled-components) and without (removing custom syntax and running against .css files)

### Proposal to fix the bug

No idea

**Comment by lipe-dev** [comment]
update:
also tried running with `./node_modules/.bin/stylelint src/layouts/index.tsx  `

**Comment by ybiquitous** [comment]
@lipe-dev Thanks for using the template! Can you share the output of `stylelint --print-config` here?

```shell
stylelint --print-config .stylelintrc.js
```

**Comment by artola** [comment]
@ybiquitous The same for me with a simple config:

stylelint: 14.1.0
stylelint-config-recommended: 6.0.0
@stylelint/postcss-css-in-js: 0.37.2
postcss: 7.0.39
postcss-syntax: 0.36.2

```js
module.exports = {
  extends: ['stylelint-config-recommended'],
  customSyntax: '@stylelint/postcss-css-in-js',
};
```
```jsx
import {css} from '@emotion/css';

export const Styled = (props) => (
  <div
    className={css`
      color: #foo;
      width: 100;
    `}
    {...props}
  />
);
```

```json
{
  "customSyntax": "@stylelint/postcss-css-in-js",
  "rules": {
    "at-rule-no-unknown": [
      true
    ],
    "block-no-empty": [
      true
    ],
    "color-no-invalid-hex": [
      true
    ],
    "comment-no-empty": [
      true
    ],
    "custom-property-no-missing-var-function": [
      true
    ],
    "declaration-block-no-duplicate-custom-properties": [
      true
    ],
    "declaration-block-no-duplicate-properties": [
      true,
      {
        "ignore": [
          "consecutive-duplicates-with-different-values"
        ]
      }
    ],
    "declaration-block-no-shorthand-property-overrides": [
      true
    ],
    "font-family-no-duplicate-names": [
      true
    ],
    "font-family-no-missing-generic-family-keyword": [
      true
    ],
    "function-calc-no-unspaced-operator": [
      true
    ],
    "function-linear-gradient-no-nonstandard-direction": [
      true
    ],
    "keyframe-declaration-no-important": [
      true
    ],
    "media-feature-name-no-unknown": [
      true
    ],
    "named-grid-areas-no-invalid": [
      true
    ],
    "no-descending-specificity": [
      true
    ],
    "no-duplicate-at-import-rules": [
      true
    ],
    "no-duplicate-selectors": [
      true
    ],
    "no-empty-source": [
      true
    ],
    "no-extra-semicolons": [
      true
    ],
    "no-invalid-double-slash-comments": [
      true
    ],
    "no-invalid-position-at-import-rule": [
      true
    ],
    "no-irregular-whitespace": [
      true
    ],
    "property-no-unknown": [
      true
    ],
    "selector-pseudo-class-no-unknown": [
      true
    ],
    "selector-pseudo-element-no-unknown": [
      true
    ],
    "selector-type-no-unknown": [
      true,
      {
        "ignore": [
          "custom-elements"
        ]
      }
    ],
    "string-no-newline": [
      true
    ],
    "unit-no-unknown": [
      true
    ]
  }
}
```

**Comment by ybiquitous** [comment]
@artola 

> postcss: 7.0.39

Stylelint 14 requires PostCSS 8. Can you update to the latest versions, please?

**Comment by artola** [comment]
@ybiquitous `postcss` is a dependency of `stylelint` and it is properly installed in the own folder. Even updating the version in my package to `8.4.4` the linter does not detect any error.

**Comment by ybiquitous** [comment]
@artola When I try the following reproduction, 1 error is reported. Can you provide more info?

`package.json`:

```json
{
  "dependencies": {
    "@stylelint/postcss-css-in-js": "0.37.2",
    "postcss": "7.0.39",
    "postcss-syntax": "0.36.2",
    "stylelint": "14.1.0",
    "stylelint-config-recommended": "6.0.0"
  }
}
```

`.stylelintrc.js`:

```js
module.exports = {
  extends: ['stylelint-config-recommended'],
  customSyntax: '@stylelint/postcss-css-in-js',
};
```

`app.jsx`:

```jsx
import React from 'react';
import {css} from '@emotion/css';

export const Styled = (props) => (
  <div
    className={css`
      color: #foo;
      width: 100;
    `}
    {...props}
  />
);
```

Run:

```console
$ ./node_modules/.bin/stylelint app.jsx

app.jsx
 7:14  ✖  Unexpected invalid hex color "#foo"  color-no-invalid-hex
```

**Comment by artola** [comment]
@ybiquitous You are right, I started a new folder with a few dependencies and there it works. I will check why there other one that is a CRA ejected is not working. I will be back with some info asap.

```
    "@babel/core": "^7.16.0",
    "@emotion/css": "^11.5.0",
    "@stylelint/postcss-css-in-js": "^0.37.2",
    "postcss": "^8.4.4",
    "postcss-syntax": "^0.36.2",
    "prettier": "^2.5.0",
    "react": "^17.0.2",
    "stylelint": "^14.1.0",
    "stylelint-config-recommended": "^6.0.0"
```

**Comment by lipe-dev** [comment]
@ybiquitous thanks for taking the time to look into it, here is the output:

```
{
  "plugins": [
    "/home/felipe/Projects/LogRock/granite/node_modules/stylelint-order/index.js",
    "/home/felipe/Projects/LogRock/granite/node_modules/stylelint-order/index.js"
  ],
  "customSyntax": "@stylelint/postcss-css-in-js",
  "rules": {
    "order/properties-order": [
      [
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "heading",
          "properties": [
            "content",
            "quotes"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "box",
          "properties": [
            "display",
            "visibility",
            "position",
            "z-index",
            "top",
            "right",
            "bottom",
            "left",
            "box-sizing",
            "grid",
            "grid-after",
            "grid-area",
            "grid-auto-columns",
            "grid-auto-flow",
            "grid-auto-rows",
            "grid-before",
            "grid-column",
            "grid-column-end",
            "grid-column-gap",
            "grid-column-start",
            "grid-columns",
            "grid-end",
            "grid-gap",
            "grid-row",
            "grid-row-end",
            "grid-row-gap",
            "grid-row-start",
            "grid-rows",
            "grid-start",
            "grid-template",
            "grid-template-areas",
            "grid-template-columns",
            "grid-template-rows",
            "flex",
            "flex-basis",
            "flex-direction",
            "flex-flow",
            "flex-grow",
            "flex-shrink",
            "flex-wrap",
            "align-content",
            "align-items",
            "align-self",
            "justify-content",
            "order",
            "width",
            "min-width",
            "max-width",
            "height",
            "min-height",
            "max-height",
            "margin",
            "margin-top",
            "margin-right",
            "margin-bottom",
            "margin-left",
            "padding",
            "padding-top",
            "padding-right",
            "padding-bottom",
            "padding-left",
            "float",
            "clear",
            "overflow",
            "overflow-x",
            "overflow-y",
            "clip",
            "zoom",
            "columns",
            "column-gap",
            "column-fill",
            "column-rule",
            "column-span",
            "column-count",
            "column-width",
            "table-layout",
            "empty-cells",
            "caption-side",
            "border-spacing",
            "border-collapse",
            "list-style",
            "list-style-position",
            "list-style-type",
            "list-style-image"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "animation",
          "properties": [
            "transform",
            "transform-box",
            "transform-origin",
            "transform-style",
            "backface-visibility",
            "perspective",
            "perspective-origin",
            "transition",
            "transition-property",
            "transition-duration",
            "transition-timing-function",
            "transition-delay",
            "animation",
            "animation-name",
            "animation-duration",
            "animation-play-state",
            "animation-timing-function",
            "animation-delay",
            "animation-iteration-count",
            "animation-direction"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "border",
          "properties": [
            "border",
            "border-top",
            "border-right",
            "border-bottom",
            "border-left",
            "border-width",
            "border-top-width",
            "border-right-width",
            "border-bottom-width",
            "border-left-width",
            "border-style",
            "border-top-style",
            "border-right-style",
            "border-bottom-style",
            "border-left-style",
            "border-radius",
            "border-top-left-radius",
            "border-top-right-radius",
            "border-bottom-right-radius",
            "border-bottom-left-radius",
            "border-color",
            "border-top-color",
            "border-right-color",
            "border-bottom-color",
            "border-left-color",
            "outline",
            "outline-color",
            "outline-offset",
            "outline-style",
            "outline-width",
            "stroke-width",
            "stroke-linecap",
            "stroke-dasharray",
            "stroke-dashoffset",
            "stroke"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "background",
          "properties": [
            "opacity",
            "background",
            "background-attachment",
            "background-clip",
            "background-color",
            "background-image",
            "background-repeat",
            "background-position",
            "background-size",
            "box-shadow",
            "fill"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "text",
          "properties": [
            "color",
            "font",
            "font-family",
            "font-size",
            "font-size-adjust",
            "font-smoothing",
            "font-stretch",
            "font-style",
            "font-variant",
            "font-weight",
            "font-emphasize",
            "font-emphasize-position",
            "font-emphasize-style",
            "letter-spacing",
            "line-height",
            "list-style",
            "text-align",
            "text-align-last",
            "text-decoration",
            "text-decoration-color",
            "text-decoration-line",
            "text-decoration-style",
            "text-indent",
            "text-justify",
            "text-overflow",
            "text-overflow-ellipsis",
            "text-overflow-mode",
            "text-rendering",
            "text-outline",
            "text-shadow",
            "text-transform",
            "text-wrap",
            "word-wrap",
            "word-break",
            "text-emphasis",
            "text-emphasis-color",
            "text-emphasis-style",
            "text-emphasis-position",
            "vertical-align",
            "white-space",
            "word-spacing",
            "hyphens",
            "src"
          ]
        },
        {
          "noEmptyLineBetween": true,
          "emptyLineBefore": "always",
          "groupName": "other",
          "properties": [
            "tab-size",
            "counter-reset",
            "counter-increment",
            "resize",
            "cursor",
            "pointer-events",
            "speak",
            "user-select",
            "nav-index",
            "nav-up",
            "nav-right",
            "nav-down",
            "nav-left"
          ]
        }
      ]
    ],
    "at-rule-no-unknown": [
      true
    ],
    "block-no-empty": [
      true
    ],
    "color-no-invalid-hex": [
      true
    ],
    "comment-no-empty": [
      true
    ],
    "custom-property-no-missing-var-function": [
      true
    ],
    "declaration-block-no-duplicate-custom-properties": [
      true
    ],
    "declaration-block-no-duplicate-properties": [
      true,
      {
        "ignore": [
          "consecutive-duplicates-with-different-values"
        ]
      }
    ],
    "declaration-block-no-shorthand-property-overrides": [
      true
    ],
    "font-family-no-duplicate-names": [
      true
    ],
    "font-family-no-missing-generic-family-keyword": [
      true
    ],
    "function-calc-no-unspaced-operator": [
      true
    ],
    "function-linear-gradient-no-nonstandard-direction": [
      true
    ],
    "keyframe-declaration-no-important": [
      true
    ],
    "media-feature-name-no-unknown": [
      true
    ],
    "named-grid-areas-no-invalid": [
      true
    ],
    "no-descending-specificity": [
      true
    ],
    "no-duplicate-at-import-rules": [
      true
    ],
    "no-duplicate-selectors": [
      true
    ],
    "no-empty-source": [
      true
    ],
    "no-extra-semicolons": null,
    "no-invalid-double-slash-comments": [
      true
    ],
    "no-invalid-position-at-import-rule": [
      true
    ],
    "no-irregular-whitespace": [
      true
    ],
    "property-no-unknown": [
      true
    ],
    "selector-pseudo-class-no-unknown": [
      true
    ],
    "selector-pseudo-element-no-unknown": [
      true
    ],
    "selector-type-no-unknown": [
      true,
      {
        "ignore": [
          "custom-elements"
        ]
      }
    ],
    "string-no-newline": [
      true
    ],
    "unit-no-unknown": [
      true
    ],
    "indentation": null,
    "linebreaks": null,
    "at-rule-empty-line-before": null,
    "at-rule-name-case": null,
    "at-rule-name-newline-after": null,
    "at-rule-name-space-after": null,
    "at-rule-semicolon-newline-after": null,
    "at-rule-semicolon-space-before": null,
    "block-closing-brace-empty-line-before": null,
    "block-closing-brace-newline-after": null,
    "block-closing-brace-newline-before": null,
    "block-closing-brace-space-after": null,
    "block-closing-brace-space-before": null,
    "block-opening-brace-newline-after": null,
    "block-opening-brace-newline-before": null,
    "block-opening-brace-space-after": null,
    "block-opening-brace-space-before": null,
    "color-hex-case": null,
    "declaration-bang-space-after": null,
    "declaration-bang-space-before": null,
    "declaration-block-semicolon-newline-after": null,
    "declaration-block-semicolon-newline-before": null,
    "declaration-block-semicolon-space-after": null,
    "declaration-block-semicolon-space-before": null,
    "declaration-block-trailing-semicolon": null,
    "declaration-colon-newline-after": null,
    "declaration-colon-space-after": null,
    "declaration-colon-space-before": null,
    "function-comma-newline-after": null,
    "function-comma-newline-before": null,
    "function-comma-space-after": null,
    "function-comma-space-before": null,
    "function-max-empty-lines": null,
    "function-parentheses-newline-inside": null,
    "function-parentheses-space-inside": null,
    "max-empty-lines": null,
    "max-line-length": null,
    "media-feature-colon-space-after": null,
    "media-feature-colon-space-before": null,
    "media-feature-name-case": null,
    "media-feature-parentheses-space-inside": null,
    "media-query-list-comma-newline-after": null,
    "media-query-list-comma-newline-before": null,
    "media-query-list-comma-space-after": null,
    "media-query-list-comma-space-before": null,
    "no-empty-first-line": null,
    "no-eol-whitespace": null,
    "no-missing-end-of-source-newline": null,
    "number-leading-zero": null,
    "number-no-trailing-zeros": null,
    "property-case": null,
    "selector-attribute-brackets-space-inside": null,
    "selector-attribute-operator-space-after": null,
    "selector-attribute-operator-space-before": null,
    "selector-attribute-quotes": null,
    "selector-combinator-space-after": null,
    "selector-combinator-space-before": null,
    "selector-descendant-combinator-no-non-space": null,
    "selector-list-comma-newline-after": null,
    "selector-list-comma-newline-before": null,
    "selector-list-comma-space-after": null,
    "selector-list-comma-space-before": null,
    "selector-max-empty-lines": null,
    "selector-pseudo-class-case": null,
    "selector-pseudo-class-parentheses-space-inside": null,
    "selector-pseudo-element-case": null,
    "string-quotes": null,
    "unicode-bom": null,
    "unit-case": null,
    "value-list-comma-newline-after": null,
    "value-list-comma-newline-before": null,
    "value-list-comma-space-after": null,
    "value-list-comma-space-before": null,
    "value-list-max-empty-lines": null
  },
  "pluginFunctions": {}
}
```

**Comment by artola** [comment]
@ybiquitous I found the tip of the iceberg. To the same code that you have in this [comment](https://github.com/stylelint/stylelint/issues/5755#issuecomment-985523206) just add to the package 1 more dependency and the the config, and then it is not reporting any error. As I said I used a CRA ejected to test a new shareable config that I am writing, as it was not showing errors, created the example with a minimal and well known config.

```
"devDependencies": {
    "babel-preset-react-app": "^10.0.0",
...
},
"babel": {
    "presets": [
      "react-app"
    ]
  }
```

**Comment by artola** [comment]
@ybiquitous When using with CRA ejected, it use `babel-preset-react-app` and when running from the command line as:

```sh
yarn stylelint Styled.jsx
```

It does not show any error, but internally the above plugin is throwing an error because there are not the `env` variables set.
Then, the problem running `stylelint` in the CLI is that the error is swallowed and not printed.

```
const env = process.env.BABEL_ENV || process.env.NODE_ENV;
...
  var isEnvDevelopment = env === 'development';
  var isEnvProduction = env === 'production';
  var isEnvTest = env === 'test';

  if (!isEnvDevelopment && !isEnvProduction && !isEnvTest) {
    throw new Error(
      'Using `babel-preset-react-app` requires that you specify `NODE_ENV` or ' +
        '`BABEL_ENV` environment variables. Valid values are "development", ' +
        '"test", and "production". Instead, received: ' +
        JSON.stringify(env) +
        '.'
    );
  }
```

**Comment by ybiquitous** [comment]
@artola Thank you. I can reproduce it:

No errors reported without `NODE_ENV`:

```console
$ node_modules/.bin/stylelint app.jsx
```

1 error reported with `NODE_ENV`:

```console
$ NODE_ENV=production node_modules/.bin/stylelint app.jsx

app.jsx
 7:14  ✖  Unexpected invalid hex color "#foo"  color-no-invalid-hex
```

I'm not sure, but is there a possible case not using `NODE_ENV`?

Anyway, it seems the following code in `@stylelint/postcss-css-in-js` suppresses any errors. I don't know the suppressing reason.

https://github.com/stylelint/postcss-css-in-js/blob/c44f6b1e221305e160169eb258a17eb00e3644c4/extract.js#L134-L136

**Comment by ybiquitous** [comment]
FYI. We are discussing in deprecating `@stylelint/postcss-css-in-js` (see https://github.com/stylelint/postcss-css-in-js/issues/225).

**Comment by artola** [comment]
@ybiquitous If as properly documented `@stylelint/postcss-css-in-js` will be deprecated and `postcss-jsx` is a proted version with no commit in 2 years, what is your recommendation as replacement?

**Comment by ybiquitous** [comment]
@artola 

> May be here also reporting the error in the `catch` will help.

Yes, it looks like a good solution.

> what is your recommendation as replacement?

#225 says, "If you use Stylelint to lint CSS-in-JS, please consider creating a syntax for the library you use". 
We have struggled with CSS-in-JS problems for a long time (see #4574).

**Comment by lipe-dev** [comment]
I tried a few of the things @artola and @ybiquitous discussed, but no luck.

I guess I'll just skip using stylelint for now.

**Comment by astone123** [comment]
@ybiquitous @lipe-dev I'm having the same issue. I added a log where the error is suppressed in the `@stylelint/postcss-css-in-js`, it looks like there's an issue inside of the `literalParser` function when it calls `parse` from `@babel/core`. This is the stack trace

```
TypeError: p is not iterable
    at <projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10658:43
    at Array.some (<anonymous>)
    at hasPlugin (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10654:18)
    at validatePlugins (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10694:9)
    at getParser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15287:5)
    at parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15232:22)
    at parser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parser/index.js:52:34)
    at parser.next (<anonymous>)
    at Function.parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parse.js:31:37)
    at parse.next (<anonymous>)
```

It looks like it has something to do with Babel plugins, but I'm not sure what. Might try opening an issue in the `@stylelint/postcss-css-in-js` repo.

**Comment by phyllisstein** [comment]
That `p is not iterable` error was causing Stylelint to crash silently for me as well. I resolved it by deleting my `.babelrc` file. As far as I can tell, Stylelint will only run when it’s completely empty or missing altogether. Even anodyne configurations caused crashes.

**Comment by DustinBrett** [comment]
Deleting `.babelrc` also fixed the issue for me of Stylelint reporting no issues while also using `postcss-css-in-js`.
